src/metabase/server/middleware/session.clj

```diff
@@ -351,21 +351,41 @@
 ;;; |                                              reset-cookie-timeout                                             |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
+(defn- check-session-timeout
+  "Asserts the [[session-timeout]] value is valid."
+  [timeout]
+  (when (some? timeout)
+    (let [{:keys [unit amount]} timeout
+          days (/ amount (case unit
+                           "seconds" (* 60 60 24)
+                           "minutes" (* 60 24)
+                           "hours"   24))
+          years (/ days 365.25)]
+      (assert (pos? amount) (tru "Session timeout amount must be positive."))
+      (assert (< years 100) (tru "Session timeout must be less than 100 years.")))))
+
 (defsetting session-timeout
   ;; Should be in the form {:amount 60 :unit "minutes"} where the unit is one of "seconds", "minutes" or "hours".
   ;; The amount is nillable.
   (deferred-tru "Time before inactive users are logged out. By default, sessions last indefinitely.")
-  :type       :json
-  :default    nil)
+  :type    :json
+  :default nil
+  :getter  (fn []
+             (let [value (setting/get-value-of-type :json :session-timeout)]
+               (check-session-timeout value)
```

Comment:
I would rather not effectively brick the instance if someone had previously set a bad value. Wouldn't this make it impossible to log in? 

That's why I was suggesting falling back to some default or something in the better rather than erroring